migrate.go

```diff
@@ -389,34 +395,74 @@ func PlanMigration(db *sql.DB, dialect string, m MigrationSource, dir MigrationD
 	return result, dbMap, nil
 }
 
-// Filter a slice of migrations into ones that should be applied.
-func ToApply(migrations []*Migration, current string, direction MigrationDirection) []*Migration {
-	var index = -1
-	if current != "" {
-		for index < len(migrations)-1 {
-			index++
-			if migrations[index].Id == current {
+// Searches the arrays for the latest elements with common ids.
+// Returns the indexes of the common id.
+func lastCommon(left []*Migration, right []*Migration) (int, int) {
+	xIndexMatch := -1
+	yIndexMatch := 0
+	for i := 0; i < len(left); i++ {
+		existingId := left[i].Id
+		for j := yIndexMatch; j < len(right); j++ {
+			if right[j].Id == existingId {
+				xIndexMatch = i
+				yIndexMatch = j
 				break
 			}
 		}
 	}
+	if xIndexMatch == -1 {
+		return -1, -1 // We never found a match; the arrays have nothing in common
```

Comment:
I don't expect this to be relevant outside of this function, so I don't think I'll add the const, unless you think it makes a big difference.